src/zope/app/component/testing.py

```diff
@@ -11,185 +11,44 @@
 # FOR A PARTICULAR PURPOSE.
 #
 ##############################################################################
-"""Base Mix-in class for Placeful Setups
-
-Also contains common test related classes/functions/objects.
 """
+Test helpers.
+"""
+from __future__ import absolute_import
 
-import os
 import zope.interface
 import zope.site.folder
-from zope.component.interfaces import IComponentLookup
-from zope.app.component.interfaces import ILocalSiteManager
-from zope.app.testing import setup
-from zope.app.testing.placelesssetup import PlacelessSetup
-
-from zope.app.testing.functional import ZCMLLayer
-from zope.traversing.api import traverse
-
-AppComponentLayer = ZCMLLayer(
-    os.path.join(os.path.split(__file__)[0], 'ftesting.zcml'),
-    __name__, 'AppComponentLayer', allow_teardown=True)
-
-class Place(object):
-
-    def __init__(self, path):
-        self.path = path
-
-    def __get__(self, inst, cls=None):
-        if inst is None:
-            return self
-
-        try:
-            # Use __dict__ directly to avoid infinite recursion
-            root = inst.__dict__['rootFolder']
-        except KeyError:
-            root = inst.rootFolder = setup.buildSampleFolderTree()
-
-        return traverse(root, self.path)
-
-
-class PlacefulSetup(PlacelessSetup):
```

Comment:
I've looked into it, and`zope.app.applicationcontrol` doesn't actually need the old `z.a.component.testing.PlacefulSetup`, it really just needed `zope.component.testing.PlacelessSetup`. The same goes for `zope.app.authentication`.

`z3c.jsontree` does appear to want all the folder setup that it used to do; it lists folder trees and compares them to on-disk canned structures. It could be argued that having the folder setup closer to where the tests are actually performed is better.